lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp

```diff
@@ -2436,6 +2436,7 @@ handleConditionalDestroys(SILValue ControlVariableAddr) {
     auto &Availability = CDElt.Availability;
 
     B.setInsertionPoint(Release);
+    B.setCurrentDebugScope(Release->getDebugScope());
```

Comment:
This makes me wonder if we should make it illegal to set the insertion point without also doing something to the scope. Should we replace setInsertionPoint with setInsertionPointAndDebugScope() ?

Reply:
@vedantk For the "let's improve SILBuilder ergonomics" task ^^

Reply:
@dcci: or should these all be fresh SILBuilderWithScopes?

Reply:
Probably we should do something like that, let me think and let's chat further.

Reply:
I proposed setInsertionPointAndDebugScope() for llvm's IRBuilder. The main sticking point was on how to update all existing uses of setInsertionPoint(). It's worth having a separate discussion about this for SILGen.